### PR TITLE
Specify the segment name that contains the reflection sections.

### PR DESCRIPTION
**Pull request by augusto2112**
Specify the segment name that contains the reflection sections in different object file formats.

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci smoke test

File: include/swift/ABI/ObjectFile.h [hunk 1]
```diff
@@ -28,6 +28,9 @@ class SwiftObjectFileFormat {
 public:
   virtual ~SwiftObjectFileFormat() {}
   virtual llvm::StringRef getSectionName(ReflectionSectionKind section) = 0;
+  virtual llvm::Optional<llvm::StringRef> getContainingSegmentName() {
```
Comment (kastiglione) [inline include/swift/ABI/ObjectFile.h:31]: I think it would be fine to name this `getSegmentName`

**Comment by augusto2112** [comment]
@swift-ci smoke test

**Review by adrian-prantl — APPROVED** [review]
This is an odd one, because MachO is the only format that even has the concept of a segment, but I can't really think of a much better interface for this either. `getReflectionSectionParentName()` does not sound any better to me.

**Pull request merged**